### PR TITLE
MSC3429: Individual room preview API

### PR DESCRIPTION
**Pull request by turt2live**
[Rendered](https://github.com/matrix-org/matrix-doc/blob/travis/msc/individual-room-summary/proposals/3429-individual-room-preview-api.md)

### CODE REVIEW

File: proposals/3429-individual-room-preview-api.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+# MSC3429: Individual room preview API
```
Comment (tulir) [inline proposals/3429-individual-room-preview-api.md:1]: This seems very similar to [MSC3266](https://github.com/matrix-org/matrix-doc/pull/3266)

Comment (turt2live) [reply proposals/3429-individual-room-preview-api.md:1]: Gah, that never came up in research for people who have tried to do this exact thing.

Comment (turt2live) [reply proposals/3429-individual-room-preview-api.md:1]: have mentioned it as an alternative

**Comment by clokep** [comment]
> I say this, because i think stripped-state (which is used in multiple places), compared to the summary mechanism (which is only used in MSC3266, and is its own mechanism), would allow for code re-use and a more standard way to approach this.

MSC3266 is based on MSC2946, which attempts to re-use the response [format of `/publicRooms`](https://matrix.org/docs/spec/client_server/r0.6.0#post-matrix-client-r0-publicrooms).

Stripped state is only used for invites/knocking IIRC.

Please use threads. :)

File: proposals/3429-individual-room-preview-api.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+not it'll have enough permission to expose the state, but might not have persisted the stripped state
+from an invite sent to a local user, thus needing to re-request it. The requesting server is still
+required to validate that the user making the call has appropriate permission to access the stripped
+state for the room.
+
+There are no particular caching recommendations, however given the information about a room rarely
+changes, the suggestion from the author is to use the same caching semantics as the room directory.
+If a room becomes non-public, it should be evicted from the cache.
+
+## Alternatives
```
Comment (turt2live) [inline proposals/3429-individual-room-preview-api.md:67]: What if we just use the spaces summary? https://github.com/matrix-org/matrix-doc/pull/3266#discussion_r686213648

**Review by KitsuneRal — CHANGES_REQUESTED** [review]
As a client author, I like #3266 shape more - as its author states, it is easier to extract data specifically for the room preview case, even if at the expense of having to add fields to yet another endpoint whenever the stripped state definition gets extended. Compared to that MSC, the current MSC looks like half-way to room peeking. I'd rather use peeking to get a richer/live preview if that provided by #3266 (or hereby) is not enough; but I don't see a lot of value in full-blown stripped state events to address a simpler preview case. The wider interface calls for unneeded questions like whether it's the same stripped state returned by `/sync` for invited rooms and if yes, should `unsigned/prev_content` also be returned - all those rather answered by the peeking MSC.

Update: copied to the relevant thread.

File: proposals/3429-individual-room-preview-api.md (hunk 2 above)
Comment (KitsuneRal) [reply proposals/3429-individual-room-preview-api.md:67]: As a client author, I like #3266 shape more - as its author states, it is easier to extract data specifically for the room preview case, even if at the expense of having to add fields to yet another endpoint whenever the stripped state definition gets extended. Compared to that MSC, the current MSC looks like half-way to room peeking. I'd rather use peeking to get a richer/live preview if that provided by #3266 (or hereby) is not enough; but I don't see a lot of value in full-blown stripped state events to address a simpler preview case. The wider interface calls for unneeded questions like whether it's the same stripped state returned by `/sync` for invited rooms and if yes, should `unsigned/prev_content` also be returned - all those rather answered by the peeking MSC.

**Pull request closed without merging**